tests/integration/test_insert_into_distributed/test.py

```diff
@@ -288,7 +288,7 @@ def test_inserts_single_replica_no_internal_replication(started_cluster):
                     "prefer_localhost_replica": "0",
                 },
             )
-        assert node2.query("SELECT count(*) FROM single_replicated").strip() == "1"
+        assert node2.query("SELECT count(*) FROM single_replicated").strip() == "0"
```

Comment:
I'm not sure what exactly this test checks, but I guess it's kind of expected that no data is inserted on exception. This test was added as a result of an attempt to fix a flaky test by splitting it into multiple tests with different settings: https://github.com/ClickHouse/ClickHouse/pull/23751